std/http/server.ts

```diff
@@ -252,9 +252,12 @@ export type HTTPOptions = Omit<Deno.ListenOptions, "transport">;
  *       req.respond({ body });
  *     }
  */
+
 export function serve(addr: string | HTTPOptions): Server {
   if (typeof addr === "string") {
-    const [hostname, port] = addr.split(":");
+    const addrParts = addr.split(":");
+    const port = parseInt(addrParts.pop()!, 10);
+    const hostname = addrParts.length ? addrParts.join(":") : undefined;
     addr = { hostname, port: Number(port) };
```

Comment:
Can we use the `URL` constructor here? It has IPv6 support.

```suggestion
    let url;
    try {
      url = new URL(`http://${addr}`);
    } catch {
      throw new TypeError("Invalid address.")
    }
    if (url.pathname || url.search || url.hash) {
      throw new TypeError("Invalid address.");
    }
    addr = { hostname: url.hostname, port: Number(url.port) };
```
Also, this should be extracted and applied to `listenAndServe()` as well.

Reply:
Will do, why did you add `http://` in `new URL`? shouldn't it be part of the addr string?

Reply:
> Will do, why did you add `http://` in `new URL`? shouldn't it be part of the addr string?

No? It wasn't before. Doesn't make sense for the protocol to be part of the address here.

Reply:
So how does the user pass the protocol? in `HttpOptions`? the user has an option to pass a string to the fn.
Can you please explain? :)

Reply:
The protocol is `http`... why would they pass a possibly different one?

Reply:
What about `https`?

Reply:
Oh sorry, `https` is `serveTLS` right?

Reply:
Another question @nayeemrmn , why should this logic be applied to `listenAndServe` as well? it always calls `serve` after all...

Reply:
> why should this logic be applied to `listenAndServe` as well? it always calls `serve` after all...

My mistake, you're right. No need to extract for `listenAndServe()`.